mmcv/fileio/file_client.py

```diff
@@ -173,25 +330,122 @@ def get(self, filepath):
+                ``filepath``. Default: 'utf-8'.
+        """
         filepath = str(filepath)
-        with open(filepath, 'r') as f:
+        with open(filepath, 'r', encoding=encoding) as f:
             value_buf = f.read()
         return value_buf
 
+    def put(self, obj: bytes, filepath: Union[str, Path]) -> None:
+        """Write data to a given ``filepath`` with 'wb' mode.
+
+        Args:
+            obj (bytes): Data to be written.
+            filepath (str or Path): Path to write data.
+        """
+        filepath = str(filepath)
+        with open(filepath, 'wb') as f:
+            f.write(obj)
+
+    def put_text(self,
+                 obj: str,
+                 filepath: Union[str, Path],
+                 encoding: str = 'utf-8') -> None:
+        """Write data to a given ``filepath`` with 'w' mode.
+
+        Args:
+            obj (str): Data to be written.
+            filepath (str or Path): Path to write data.
+            encoding (str, optional): The encoding format used to open the
+                `filepath`. Default: 'utf-8'.
+        """
+        filepath = str(filepath)
+        with open(filepath, 'w', encoding=encoding) as f:
+            f.write(obj)
+
+    def remove(self, filepath: Union[str, Path]) -> None:
+        """Remove a file.
+
+        Args:
+            filepath (str or Path): Path to be removed.
+        """
+        filepath = str(filepath)
+        os.remove(filepath)
+
+    def exists(self, filepath: Union[str, Path]) -> bool:
+        """Check a ``filepath`` whether exists.
+
+        Args:
+            filepath (str or Path): Path to be checked whether exists.
+        """
+        return osp.exists(str(filepath))
+
+    def isdir(self, filepath: Union[str, Path]) -> bool:
+        """Check a ``filepath`` whether it is a directory.
+
+        Args:
+            filepath (str or Path): Path to be checked whether it is a
+                directory.
+        """
+        return osp.isdir(str(filepath))
+
+    def isfile(self, filepath: Union[str, Path]) -> bool:
+        """Check a ``filepath`` whether it is a file.
+
+        Args:
+            filepath (str or Path): Path to be checked whether it is a file.
+        """
+        return osp.isfile(str(filepath))
+
+    def concat_paths(self, filepath: Union[str, Path],
+                     *filepaths: Union[str, Path]) -> str:
+        """Concatenate all filepaths.
+
+        Join one or more filepath components intelligently. The return value
+        is the concatenation of filepath and any members of *filepaths.
+
+        Args:
+            filepath (str or Path): Path to be concatenated.
+        """
+        filepath = str(filepath)
+        filepaths = [str(path) for path in filepaths]
+        return osp.join(filepath, *filepaths)
+
+    def _release_resource(self, filepath: str) -> None:
+        """Do nothing in order to unify API."""
+        pass
+
+    def _get_local_path(self, filepath: str) -> str:
```

Comment:
shall we make it get_local_path rather than _get_local_path

Reply:
One is that the method does not use the content manager, which does not destroy the file on exit, and the other is to provide a way for the user to download the file locally without destroying it, although this method is not recommended